common/src/main/java/io/pravega/common/concurrent/ExecutorServiceHelpers.java

```diff
@@ -86,8 +100,24 @@ public Thread newThread(Runnable r) {
      * @return A new executor service.
      */
     public static ScheduledExecutorService newScheduledThreadPool(int size, String poolName) {
+        return newScheduledThreadPool(size, poolName, Thread.NORM_PRIORITY);
+    }
+
+    /**
+     * Creates a new ScheduledExecutorService that will use daemon threads with specified priority and names.
+     *
+     * @param size The number of threads in the threadpool
+     * @param poolName The name of the pool (this will be printed in logs)
+     * @param threadPriority The priority to be assigned to the threads
+     * @return A new executor service.
+     */
+    public static ScheduledExecutorService newScheduledThreadPool(int size, String poolName, Integer threadPriority) {
+
+        ThreadFactory threadFactory =
```

Comment:
can we remove this redundant null check? The user would invoke `newScheduledThreadPool(int size, String poolName)` if the priority is default.

Reply:
Yeah, but since it is a public method, we can't assume that's always true. Besides, having the check makes the intention explicit.